### DISCUSSION TITLE
Writing a serialized String but padding the underlying `Vec` with null bytes.

### DISCUSSION THREAD

**Discussion by AbstractUmbra** [jam1garner/binrw#189] [Q&A]
Hello, furthering on from #187 I now have the following implementation:
```rs
#[binrw]
#[brw(little, import(version: [u8; 4], resource_offset: u32))]
#[derive(Default, Debug, Eq, PartialEq)]
struct ErfResource {
    #[br(temp, calc = *version.last().unwrap())]
    #[bw(ignore)]
    version_char: u8,

    #[br(count=if version_char == 48 { 16 } else { 32 }, try_map = String::from_utf8)]
    #[bw(map = |s| s.as_bytes().to_vec())]
    reference: String,

    id: u32,
    r#type: u32,
    #[br(seek_before = SeekFrom::Start(resource_offset as u64), restore_position)]
    offset: u32,
    #[br(restore_position)]
    size: u32,
    #[br(ignore)]
    data: Option<InnerErfResource>,
}
```

However I noticed that the `reference` field will correctly read and parse into a String, but when it goes to be written it will not have the correct level of padding.
I'm wondering if there is a way to pad this String with `\0` to fit the request size, either during the write here or in some process.

Any help is appreciated!

**Comment by csnover** [comment]
You should be able to just use `#[bw(pad_size_to = {expr})]` to pad to the minimum required size. Note this will not truncate the output; you would also need to do that if the field size is fixed. Finally, it is not necessary to call `to_vec`; `#[bw(map = String::as_bytes)]` is sufficient.
